osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/repository/DbEncryptedSecretRepository.java

```diff
@@ -30,6 +32,19 @@ List<DbEncryptedSecret> findSecrets(
       @Param("secretType") SecretType secretType,
       @Param("secretStatus") SecretStatus secretStatus);
 
+  @Query(
+      value =
+          "SELECT es FROM DbEncryptedSecret es "
+              + "JOIN FETCH es.encryptionKeyReference "
+              + "WHERE es.deviceIdentification = :deviceIdentification "
+              + "AND es.secretType IN (:secretTypes) "
+              + "AND es.secretStatus = :secretStatus "
+              + "ORDER BY es.creationTime DESC, es.id DESC")
+  List<DbEncryptedSecret> findSecrets(
```

Comment:
The old method with a single secretType argument is now only called from ShipmentFileHandlingSteps while looping through a list of secretTypes.  So with a little refactoring this method can be removed